cmd/syslog/syslog.go

```diff
@@ -20,14 +23,37 @@ func main() {
 	server.SetFormat(syslog.RFC3164)
 	server.SetHandler(handler)
 
+	// check if syslog unix socket is already there then delete it.
+	if _, err := os.Stat(listenAddress); err == nil {
+		if err := os.Remove(listenAddress); err != nil {
+			log.Fatalf("error removing syslog socket %s: %s", listenAddress, err)
+		}
+	}
+
 	if err := server.ListenUnixgram(listenAddress); err != nil {
-		log.Fatal(err)
+		log.Fatalf("failed to listen to syslog unix socket %s, err %s", listenAddress, err)
 	}
 
+	defer os.Remove(listenAddress)
```

Comment:
Also log any failure from the call to remove -- might help our future selves to understand any pattern where the file does not get unlinked cleanly.